edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/BridgeConfig.cs

```diff
@@ -1,15 +1,158 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Hub.Core.Config
 {
+    using System;
     using System.Collections.Generic;
+    using System.Linq;
+    using Newtonsoft.Json;
 
     /// <summary>
     /// Domain object that represents Bridge configuration for MQTT Broker.
     ///
     /// This object is being constructed from the EdgeHub twin's desired properties.
     /// See <see cref="EdgeHubDesiredProperties"/> for DTO.
     /// </summary>
-    public class BridgeConfig : List<string>
+    public class BridgeConfig : List<Bridge>, IEquatable<BridgeConfig>
     {
+        public bool Equals(BridgeConfig other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return Enumerable.SequenceEqual(this, other);
+        }
+
+        public override bool Equals(object obj)
+            => this.Equals(obj as BridgeConfig);
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hash = 17;
+                hash = this.Aggregate(hash, (acc, item) => (acc * 31 + item.GetHashCode()));
+                return hash;
+            }
+        }
+    }
+
+    public class Bridge : IEquatable<Bridge>
+    {
+        [JsonConstructor]
+        public Bridge(string endpoint, IList<Settings> settings)
+        {
+            this.Endpoint = endpoint;
+            this.Settings = settings;
+        }
+
+        [JsonProperty("endpoint", Required = Required.Always)]
+        public string Endpoint { get; }
+
+        [JsonProperty("settings", Required = Required.Always)]
```

Comment:
In the doc with the schema changed I asked if we should actually use subscriptions or topics because we might want to add other settings which are not related to topics. Let’s sync about which name to use

Reply:
Discussed offline. We can change it later if needed.